src/test/java/org/example/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,150 @@
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void saveBook_ValidCreateBookRequestDto_ShouldReturnBookDto() throws Exception {
+        BookDto expected = getBookDto();
+        String jsonRequest = objectMapper.writeValueAsString(getCreateBookRequestDto());
+
+        mockMvc.perform(post("/books")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is(200))
+                .andExpect(jsonPath("$.author", is(expected.getAuthor())))
+                .andExpect(jsonPath("$.price").value(expected.getPrice()))
+                .andExpect(jsonPath("$.title", is(expected.getTitle())))
+                .andExpect(jsonPath("$.coverImage", is(expected.getCoverImage())))
+                .andExpect(jsonPath("$.description", is(expected.getDescription())))
+                .andExpect(jsonPath("$.isbn", is(expected.getIsbn())));
+    }
+
+    @Test
+    @Sql(scripts = "classpath:database/add-book-to-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/remove-all-books-from-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Verify getAll() method works")
+    @WithMockUser(username = "user", roles = "USER")
+    public void getAll_ValidData_ShouldReturnBookDtos() throws Exception {
+        when(bookService.findAll(any(Pageable.class))).thenReturn(List.of(getBookDto()));
+        mockMvc.perform(get("/books")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size()").value(1));
+    }
+
+    @Test
+    @Sql(scripts = "classpath:database/add-book-to-books-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = "classpath:database/remove-all-books-from-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Verify getBookById() method works")
+    @WithMockUser(username = "user", roles = "USER")
+    public void getBookById_ValidId_ShouldReturnBookDto() throws Exception {
+        long bookId = 1L;
+        BookDto expected = getBookDto();
+
+        mockMvc.perform(get("/books/" + bookId))
+                .andExpect(status().is(200))
+                .andExpect(jsonPath("$.author", is(expected.getAuthor())))
+                .andExpect(jsonPath("$.price").value(expected.getPrice().doubleValue()))
+                .andExpect(jsonPath("$.title", is(expected.getTitle())))
+                .andExpect(jsonPath("$.coverImage", is(expected.getCoverImage())))
+                .andExpect(jsonPath("$.description", is(expected.getDescription())))
+                .andExpect(jsonPath("$.isbn", is(expected.getIsbn())));
+    }
+
+    @Test
+    @Sql(scripts = "classpath:database/add-book-to-books-table.sql",
```

Comment:
suggestion, you can use all Sql scripts under the class
Also, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD - by default, so you can avoid declaring this phase